service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java

```diff
@@ -66,25 +69,44 @@ public KubernetesServiceDiscovery(String serviceName, ServiceDiscoveryConfig con
 
     @Override
     public Uni<List<ServiceInstance>> fetchNewServiceInstances(List<ServiceInstance> previousInstances) {
-        Uni<List<Endpoints>> endpointsUni = Uni.createFrom().emitter(
+        Uni<Map<Endpoints, List<Pod>>> endpointsUni = Uni.createFrom().emitter(
                 emitter -> {
                     vertx.executeBlocking(future -> {
-                        List<Endpoints> endpoints = new ArrayList<>();
+                        Map<Endpoints, List<Pod>> items = new HashMap<>();
+                        List<Pod> backendPods = new ArrayList<>();
```

Comment:
it seems to me we're collecting pods for an  endpoint in this collection, shouldn't it be initialized in the for loop in line 81?

Reply:
Yes, good catch. I refactored to have it in the last commit